pilot/user/atlas/container.py

```diff
@@ -20,10 +20,39 @@
 from pilot.util.filehandling import write_file
 # import pilot.info.infoservice as infosys
 
+# imports for get_payload_proxy()
+from pilot.util import https
+import traceback
+
 import logging
 logger = logging.getLogger(__name__)
 
 
+def get_payload_proxy(proxy_outfile_name, voms_role='atlas'):
+    try:
+        # it assumes that https_setup() was done already
+        res = https.request('{pandaserver}/server/panda/getProxy'.format(pandaserver=config.Pilot.pandaserver),
+                            data={'role': voms_role})
+
+        if res is None:
+            logger.error("Unable to get proxy with role '%s' from panda server" % voms_role)
+            return False
+
+        if res['StatusCode'] != 0:
+            logger.error("When get proxy with role '%s' panda server returned: %s" % (voms_role, res['errorDialog']))
+            return False
+
+        proxy_contents = res['userProxy']
+        _outfile = open(proxy_outfile_name, 'w')
```

Comment:
Can you please use the write_file() function from pilot.util.filehandling here? (it's already imported)

Reply:
Ok, done.